components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/RestApiAdminUtils.java

```diff
@@ -74,6 +77,38 @@ public static boolean isBlockConditionAccessibleToUser(String user, BlockConditi
                 .equals(userTenantDomain);
     }
 
+    /**
+     * Validate the required properties of Custom Rule Policy
+     *
+     * @param customRuleDTO custom rule object to check
+     * @param httpMethod    HTTP method of the request
+     */
+    public static void validateCustomRuleRequiredProperties(CustomRuleDTO customRuleDTO, String httpMethod)
+            throws APIManagementException {
+
+        String propertyName;
+
+        //policyName property is validated only for POST request
+        if (httpMethod.equalsIgnoreCase(APIConstants.HTTP_POST)) {
```

Comment:
What is the ```httpMethod``` we check here? can this be null or empty?

Reply:
This validation method will be called by both POST and PUT request. For POST request, if the policyName is blank, we need to throw and exception. But for PUT request, since the policyName value is overridden using the existing policyName value, we need to skip this validation check of policyName.